utils/docker/Dockerfile.el.9

```diff
@@ -0,0 +1,206 @@
+            branch="${repo#*@}";                                          \
+            repo="${repo%@*}";                                            \
+            if [[ $branch = *:* ]]; then                                  \
+                build_number="${branch#*:}";                              \
+                branch="${branch%:*}";                                    \
+            fi;                                                           \
+        fi;                                                               \
+        echo -e "[$repo:$branch:$build_number]\n\
+name=$repo:$branch:$build_number\n\
+baseurl=${JENKINS_URL}job/daos-stack/job/$repo/job/$branch/$build_number/artifact/artifacts/el9/\n\
+enabled=1\n\
+gpgcheck=False\n" >> /etc/yum.repos.d/$repo:$branch:$build_number.repo;   \
+        cat /etc/yum.repos.d/$repo:$branch:$build_number.repo; \
+        dnf -y repolist; \
+        dnf --disablerepo=\* --enablerepo=$repo:$branch:$build_number makecache; \
+    done
+
+# Install OS updates and package.  Include basic tools and daos dependencies
+COPY ./utils/scripts/install-el9.sh /tmp/install.sh
+RUN chmod +x /tmp/install.sh && dnf -y upgrade && /tmp/install.sh && dnf -y clean all && \
+    rm -f /tmp/install.sh
+
+ARG UID=1000
+
+# Add DAOS users
+RUN useradd --no-log-init --uid $UID --user-group --create-home --shell /bin/bash \
+            --home /home/daos daos_server
+RUN echo "daos_server:daos_server" | chpasswd
+RUN useradd --no-log-init --user-group --create-home --shell /bin/bash daos_agent
+RUN echo "daos_agent:daos_agent" | chpasswd
+RUN echo "daos_server ALL=(root) NOPASSWD: ALL" >> /etc/sudoers.d/daos_sudo_setup
+
+# Create directory for DAOS backend storage
+RUN mkdir -p /opt/daos /mnt/daos /var/run/daos_server /var/run/daos_agent /home/daos/pre /home/daos/daos &&   \
+    chown -R daos_server.daos_server /opt/daos /mnt/daos /var/run/daos_server /home/daos &&  \
+    chown daos_agent.daos_agent /var/run/daos_agent
+
+ARG QUICKBUILD=false
+ARG QUICKBUILD_DEPS
+
+RUN if $QUICKBUILD; then                                          \
+        echo "Installing: $QUICKBUILD_DEPS";                      \
+        echo "$QUICKBUILD_DEPS" | sed -e '/^$/d' | tr '\n' '\0' | \
+          xargs -0 dnf -y install;                                \
+        dnf -y clean all;                                            \
+    fi
+
+ARG BULLSEYE
+RUN if [ "x$BULLSEYE" != "x" ]; then \
+      curl ${JENKINS_URL}job/daos-stack/job/tools/job/master/lastSuccessfulBuild/artifact/bullseyecoverage-linux.tar \
+        --retry 10 --retry-max-time 60 --silent --show-error -o bullseye.tar; \
+      mkdir -p bullseye; \
+      tar -C bullseye --strip-components=1 -xf bullseye.tar; \
+      pushd bullseye; \
+        ./install --quiet --key "${BULLSEYE}" --prefix /opt/BullseyeCoverage; \
+      popd; \
+      rm -f bullseye.tar; \
+      rm -rf bullseye; \
+    fi
+
```

Comment:
And this.

Reply:
Bullseye code is tested in at least one other stage, when it's enabled.  That we don't have it enabled all of the time is a pity.

But again, I don't want it falling on me again to have to resurrect all of this when somebody complains that Bullseye is not working on EL9.